libraries/networking/src/AssetRequest.cpp

```diff
@@ -0,0 +1,78 @@
+//
+//  AssetRequest.cpp
+//
+//  Created by Ryan Huffman on 2015/07/24
+//  Copyright 2015 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+#include "AssetRequest.h"
+
+#include <algorithm>
+
+#include <QThread>
+
+#include "AssetClient.h"
+#include "NodeList.h"
+
+
+AssetRequest::AssetRequest(QObject* parent, QString hash) :
+    QObject(parent),
+    _hash(hash)
+{
+}
+
+void AssetRequest::start() {
+    if (QThread::currentThread() != thread()) {
+        QMetaObject::invokeMethod(this, "start", Qt::AutoConnection);
+        //(&AssetRequest::start)
+        return;
+    }
+
+    if (_state == NOT_STARTED) {
+        _state = WAITING_FOR_INFO;
+
+        auto assetClient = DependencyManager::get<AssetClient>();
+        assetClient->getAssetInfo(_hash, [this](bool success, AssetInfo info) {
+            _info = info;
+            _data.resize(info.size);
+            const DataOffset CHUNK_SIZE = 1024;
```

Comment:
We should remove this or increase it by a bunch now that the ordered messaging is working, no?